content/post/asdf.md

```diff
@@ -0,0 +1,250 @@
+However, there is only one maintained software library written in Python.
+While historically ASDF targeted astronomers, it is actually abstracted away
+from any scientific domain and is completely versatile. ASDF features:
+
+* Transparent, automatic, on-the-fly compression and decompression with zlib, bzip2 or lz4.
+* YAML header with binary blocks appended to the tail. All the perks of using YAML are preserved.
+* Uncompressed tensors can be [memory mapped](https://en.wikipedia.org/wiki/Memory-mapped_file) so that the operating memory consumption is very low with tiny performance penalty for sequential read and write. A killer feature if your tensors are big.
+* Data structure can be validated with YAML schemas.
+* Python and numpy arrays are first-class citizens.
+* The tagging mechanism allows to extend for new binary data types easily. Though it is rarely needed in practice.
+* The schema is there but is completely optional.
+
+## Code examples
+
+By default, ASDF reads tensors from disk lazily upon the first reference.
+Thus the opening of an ASDF file is very fast, and it is easy to quickly introspect.
+The contents are always a Python `dict` and they are placed into the `tree` attribute.
+
+```python
+import asdf
+
+with asdf.open("file.asdf") as f:
+    print(f.tree)
+```
+
+Let's create a new ASDF file and see what's inside.
+
+```python
+import io
+
+import asdf
+import numpy
+
+tensor = numpy.ones(10, dtype="int32")
+buffer = io.BytesIO()
+asdf.AsdfFile(tree={
+    "tensor1": tensor,
+    "tensor2": tensor,
+    "meta": "data",
+    "int": 100
+}).write_to(buffer)
+print(buffer.getvalue().decode("utf-8", errors="backslashreplace"))
+```
+
+We should see
+
+```
+#ASDF 1.0.0
+#ASDF_STANDARD 1.2.0
+%YAML 1.1
+%TAG ! tag:stsci.edu:asdf/
+--- !core/asdf-1.1.0
+asdf_library: !core/software-1.0.0 {author: Space Telescope Science Institute, homepage: 'http://github.com/spacetelescope/asdf',
+  name: asdf, version: 2.1.0}
+history:
+  extensions:
+  - !core/extension_metadata-1.0.0
+    extension_class: asdf.extension.BuiltinExtension
+    software: {name: asdf, version: 2.1.0}
+int: 100
+meta: data
+tensor1: !core/ndarray-1.0.0
+  source: 0
+  datatype: int32
+  byteorder: little
+  shape: [10]
+tensor2: !core/ndarray-1.0.0
+  source: 0
+  datatype: int32
+  byteorder: little
+  shape: [10]
+...
+\xd3BLK0(((\x9dZ\x82\xde\xf2_\x8f\x83<B\xaa\xa4g\xde#ASDF BLOCK INDEX
+%YAML 1.1
+--- [615]
+...
+```
+
+Tree items which are not binary - "meta" and "int" - have been written inline in YAML.
+ASDF has been intelligent enough to serialize only one copy of the array.
+It is placed at the end of the file and forms a "block". `!core/ndarray-1.0.0`
+is a tag name which identifies the built-in tensor data type. "source" field
+references the block containing the array elements.
+
+It's also very cool that the block offset indexes table is placed at the bottom
+of the file, so that it is possible to append without rewriting the whole file.
```

Comment:
Do you actually take advantage of this?

Reply:
There is a place where we perform a merge operation over asdf files, I think so